app/src/org/commcare/views/widgets/ClearableWidget.java

```diff
@@ -0,0 +1,39 @@
+package org.commcare.views.widgets;
+
+import android.content.Context;
+import android.view.LayoutInflater;
+import android.widget.Button;
+
+import org.commcare.dalvik.R;
+import org.javarosa.form.api.FormEntryPrompt;
+
+/**
+ * A utility for all the widgets that want to show a clear button.
+ * This class will provide the code for creation of clearButton using {@link ClearableWidget#setupClearButton(Context, String)}.
+ * Subclasses will need to add the button to the view by themselves.
+ *
+ * See {@link SelectOneWidget} for example.
+ * @author $|-|!˅@M
+ */
+public abstract class ClearableWidget extends QuestionWidget {
+
+    protected Button clearButton;
+
+    public ClearableWidget(Context context, FormEntryPrompt p) {
+        super(context, p);
+    }
+
+    public ClearableWidget(Context context, FormEntryPrompt p, boolean inCompactGroup) {
+        super(context, p, inCompactGroup);
+    }
+
+    protected void setupClearButton(Context context, String text, int visibility) {
```

Comment:
@ShivamPokhriyal The way we are using it now, it makes more sense for this to be simply a static utilitiy method rather than a member of the Widget Classes. Thoughts ?

Reply:
Yeah, I'm not really sure if this subclassing is doing any good here, since the actual addition is happening inside the view only.